platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -475,12 +477,12 @@ def _dom_capability_detect(self):
                 sfpd_obj = sff8436Dom()
                 if sfpd_obj is None:
                     return None
-                qsfp_option_value_raw = self._read_eeprom_specific_bytes(QSFP_OPTION_VALUE_OFFSET, QSFP_OPTION_VALUE_WIDTH)
-                if qsfp_option_value_raw is not None:
-                    optional_capability = sfpd_obj.parse_option_params(qsfp_option_value_raw, 0)
-                    self._sfp_capability.dom_tx_disable_supported = optional_capability['data']['TxDisable']['value'] == 'On'
+
+                optional_capability = sfpd_obj.parse_option_params(qsfp_option_value_raw, 0)
+                self._sfp_capability.dom_tx_disable_supported = optional_capability['data']['TxDisable']['value'] == 'On'
                 dom_status_indicator = sfpd_obj.parse_dom_status_indicator(qsfp_version_compliance_raw, 1)
                 self._sfp_capability.qsfp_page3_available = dom_status_indicator['data']['FlatMem']['value'] == 'Off'
+                self._sfp_capability.dom_detect_finished = True
```

Comment:
how about putting this line to the end of this function?

Reply:
I think it will override the value in some cases.